src/go/configgenerator/cluster_generator.go

```diff
@@ -122,7 +122,7 @@ func addDnsResolversToClusters(dnsResolverAddresses string, clusters []*clusterp
 func makeMetadataCluster(serviceInfo *sc.ServiceInfo) (*clusterpb.Cluster, error) {
 	scheme, hostname, port, _, err := util.ParseURI(serviceInfo.Options.MetadataURL)
 	if err != nil {
-		return nil, err
+		return nil, fmt.Errorf("fail to parse metadata cluster URI: %v", err)
```

Comment:
Should we also print the problem URL for easy debug?

Reply:
Done. I added this logic to the shared `ParseURI` function to reduce code duplication.